pcap_to_node_pcap/main.cpp

```diff
@@ -0,0 +1,545 @@
+	}
+	else if (method == SPLIT_BY_PACKET_COUNT)
+	{
+		int paramAsInt = (paramWasSet ? atoi(param) : 0);
+		splitter = new PacketCountSplitter(paramAsInt);
+	}
+	else if (method == SPLIT_BY_IP_CLIENT)
+	{
+		int paramAsInt = (paramWasSet ? atoi(param) : SplitterWithMaxFiles::UNLIMITED_FILES_MAGIC_NUMBER);
+		splitter = new ClientIPSplitter(paramAsInt);
+	}
+	else if (method == SPLIT_BY_IP_SERVER)
+	{
+		int paramAsInt = (paramWasSet ? atoi(param) : SplitterWithMaxFiles::UNLIMITED_FILES_MAGIC_NUMBER);
+		splitter = new ServerIPSplitter(paramAsInt);
+	}
+	else if (method == SPLIT_BY_SERVER_PORT)
+	{
+		int paramAsInt = (paramWasSet ? atoi(param) : SplitterWithMaxFiles::UNLIMITED_FILES_MAGIC_NUMBER);
+		splitter = new ServerPortSplitter(paramAsInt);
+	}
+	else if (method == SPLIT_BY_2_TUPLE)
+	{
+		int paramAsInt = (paramWasSet ? atoi(param) : SplitterWithMaxFiles::UNLIMITED_FILES_MAGIC_NUMBER);
+		splitter = new TwoTupleSplitter(paramAsInt);
+	}
+	else if (method == SPLIT_BY_5_TUPLE)
+	{
+		int paramAsInt = (paramWasSet ? atoi(param) : SplitterWithMaxFiles::UNLIMITED_FILES_MAGIC_NUMBER);
+		splitter = new FiveTupleSplitter(paramAsInt);
+	}
+	else if (method == SPLIT_BY_BPF_FILTER)
+	{
+		splitter = new BpfCriteriaSplitter(std::string(param));
+	}
+	else
+		EXIT_WITH_ERROR("Unknown method '%s'", method.c_str());
+
+
+	// verify splitter param is legal, otherwise return an error
+	std::string errorStr;
+	if (!splitter->isSplitterParamLegal(errorStr))
+	{
+		EXIT_WITH_ERROR("%s", errorStr.c_str());
+	}
+
+	// prepare the output file format: /requested-path/original-file-name-[4-digit-number-starting-at-0000].pcap
+	std::string outputPcapFileName = outputPcapDir + std::string(1, SEPARATOR) + getFileNameWithoutExtension(inputPcapFileName) + "-";
+
+	// open a pcap file for reading
+	PcapFileReaderDevice reader(inputPcapFileName.c_str());
+
+	if (!reader.open())
+	{
+		EXIT_WITH_ERROR("Error opening input pcap file\n");
+	}
+
+	// set a filter if provided
+	if (filter != "")
+	{
+		if (!reader.setFilter(filter))
+			EXIT_WITH_ERROR("Couldn't set filter '%s'", filter.c_str());
+	}
+
+	printf("Started...\n");
+
+	int packetCountSoFar = 0;
+	int numOfFiles = 0;
+	RawPacket rawPacket;
+
+	// prepare a map of file number to PcapFileWriterDevice
+	std::map<int, PcapFileWriterDevice*> outputFiles;
+
+	// read all packets from input file, for each packet do:
+	while (reader.getNextPacket(rawPacket))
+	{
+		// parse the raw packet into a parsed packet
+		Packet parsedPacket(&rawPacket);
+
+		std::vector<int> filesToClose;
+
+		// call the splitter to get the file number to write the current packet to
+		int fileNum = splitter->getFileNumber(parsedPacket, filesToClose);
+		std::string ipVal = "";
+		if (method == SPLIT_BY_IP_CLIENT) {
+			if (fileNum != 0) {
+				ipVal = "-" + getIP(parsedPacket, true);
+				hyphenIP(ipVal);
+			}
+			else {
+				// file number 0 is for various misceallaneous packets (ping, ARP, etc.)
+				ipVal = "-miscellaneous";

```

Comment:
I thought for the PR that they wouldn't want to just see ignore, so I put in miscellaneous. I updated file_drop and will include it in my update PR.